pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -123,6 +133,30 @@ func (p *LaunchTemplateProvider) Get(ctx context.Context, constraints *v1alpha1.
 	return launchTemplates, nil
 }
 
+const (
+	defaultMetadataOptionsHTTPEndpoint            = ec2.LaunchTemplateInstanceMetadataEndpointStateEnabled
+	defaultMetadataOptionsHTTPProtocolIpv6        = ec2.LaunchTemplateInstanceMetadataProtocolIpv6Disabled
+	defaultMetadataOptionsHTTPPutResponseHopLimit = 2
+	defaultMetadataOptionsHTTPTokens              = ec2.LaunchTemplateHttpTokensStateRequired
+)
+
+func (p *LaunchTemplateProvider) getMetadataOptions(options *v1alpha1.MetadataOptions) launchTemplateMetadataOptions {
```

Comment:
Consider attaching this behavior to the constraints object itself. 

constraints.GetMetadataOptions()